robottelo/config/base.py

```diff
@@ -514,6 +514,43 @@ def validate(self):
         return validation_errors
 
 
+class GCESettings(FeatureSettings):
+    """Google Compute Engine settings definitions."""
+
+    def __init__(self, *args, **kwargs):
+        super(GCESettings, self).__init__(*args, **kwargs)
+        self.project_id = None
+        self.client_email = None
+        self.cert_path = None
+        self.zone = None
+
+    def read(self, reader):
+        """Read GCE settings."""
+        self.project_id = reader.get('gce', 'project_id')
+        self.client_email = reader.get('gce', 'client_email')
+        self.cert_path = reader.get('gce', 'cert_path')
+        self.zone = reader.get('gce', 'zone')
+
+    def validate(self):
+        """Validate GCE settings."""
+        valid_cert_path = '/usr/share/foreman/'
+        valid_zones = [
+            'us-east1-b', 'us-east1-c', 'us-east1-d',
+            'us-east4-a', 'us-east4-b', 'us-east4-c']
+        validation_errors = []
+        if not all(self.__dict__.values()):
+            validation_errors.append(
+                'All [gce] {} options must be provided'.format(self.__dict__.keys()))
+        if not str(self.cert_path).startswith(valid_cert_path):
+            validation_errors.append(
+                '[gce] cert_path - cert should be available '
+                'from satellites {}'.format(valid_cert_path))
+        if str(self.zone).startswith('us') and self.zone not in valid_zones:
```

Comment:
```suggestion
        if self.zone not in valid_zones:
```
I think above is fine to validate the zone .. as `True` will only return on complete match  else it would remain `False`

Reply:
There are many zones, so I don't want to get all of them stored under valid_zones. But as we tend to use more of us zones, I kept validation for us zones only.

Reply:
@jyejare Thanks , But this `self.zone not in valid_zones` validation will auto takecare the `us` one because you only have these in your `valid_zones`

Reply:
But what if somebody changes it and give something other than 'us'. We won't check for any zones other than 'us' ones.
If I wont give this condition 'if str(self.zone).startswith('us')' the test will fail there itself saying the zone is not in valid zones, though it will be a valid zone.

Reply:
on fail it would gives the clear hint around valid zones  `[gce] zone - The zone should be one of {}'.format(valid_zones))` .... Am i missing something ?

Reply:
@vijay8451 , As I said in my first comment that there are many zones and as many as 20-25, but I have this validation just for 'us' based zones.

I don't want to keep the 20-25 list of zones in valid-zones as we won't be using or rarely use other zones than 'us' ones.